.github/workflows/install.yaml

```diff
@@ -0,0 +1,71 @@
+name: Install Script
+on:
+  push:
+    branches: [release-1.22]
+    paths:
+      - "channel.yaml"
+      - "install.sh"
+      - "tests/vagrant/install/**"
+  pull_request:
+    branches: [release-1.22]
+    paths:
+      - "install.sh"
+      - "tests/vagrant/install/**"
+  workflow_dispatch: {}
+jobs:
+  test:
+    name: "Smoke Test"
+    runs-on: macos-10.15
+    timeout-minutes: 40
+    strategy:
+      fail-fast: false
+      matrix:
+        vm: [centos-7, centos-8, fedora-coreos, opensuse-leap, opensuse-microos, ubuntu-focal]
+        include:
+          - {channel: latest, vm: centos-8}
+          - {channel: latest, vm: ubuntu-focal}
+          - {channel: latest, vm: opensuse-leap}
+          - {channel: testing, vm: centos-8}
+          - {channel: testing, vm: opensuse-leap}
+          - {channel: testing, vm: ubuntu-focal}
+      max-parallel: 2
+    defaults:
+      run:
+        working-directory: tests/vagrant/install/${{ matrix.vm }}
+    steps:
+      - name: "Checkout"
+        uses: actions/checkout@v2
+        with: {fetch-depth: 1}
+      - name: "Set target k3s version"
+        run: |
+          echo "INSTALL_K3S_COMMIT=${GITHUB_SHA}" >> $GITHUB_ENV
```

Comment:
For k3s installation I could either install from commit or build the artifacts beforehand then mount them (cgroup.yaml uses this approach). Still, I am not sure what should be the best approach.

Reply:
I see the `git ls-remote` approach working on CI tests, I'll leave it that way for the moment.